File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        if (k <= 1) {
            return 0;
        }

        int result = 0, calc = 1, lo = 0;

        for (int hi = 0; hi < nums.length; hi++) {
            calc *= nums[hi];

            while (calc >= k) {
                calc /= nums[lo];
                lo++;
            }

            result += hi - lo + 1;
        }

        return result;
    }
}

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k <= 1:
            return 0

        result = 0
        calc = 1
        lo = 0

        for hi in range(len(nums)):
            calc *= nums[hi]

            while calc >= k:
                calc //= nums[lo]
                lo += 1

            result += hi - lo + 1

        return result